cirq/_compat.py

```diff
@@ -459,14 +455,36 @@ def find_spec(self, fullname: str, path: Any = None, target: Any = None) -> Any:
 
         # find the corresponding spec in the new structure
         if fullname == self.old_module_name:
-            spec = self.new_module_spec
+            # this is the first time the deprecated module is being found
+            # which means that the new parent needs to be found first and under
+            # the new parent's path, we should be able to find the new name of
+            # the deprecated module
+            # this code is heavily inspired by importlib.util.find_spec
+            parent_name = new_fullname.rpartition('.')[0]
+            if parent_name:
+                parent = __import__(parent_name, fromlist=['__path__'])
+                # note that compared to importlib.util.find_spec we don't handle
+                # AttributeError here because it is not expected to happen in case
+                # of a DeprecatedModuleLoader - the new parent should exist and be
+                # a proper package
+                parent_path = parent.__path__
+            else:
+                parent_path = None
+            spec = self.finder.find_spec(new_fullname, parent_path, None)
```

Comment:
Thanks for looking into this.  I didn't realize it would get this involved.  Now I see the core challenge here with finding the spec while bypassing the wrapper.  This solution works.

Reply:
Awesome! Thank you for the detailed, thorough review.